app/services/irs_attempts_api/tracker.rb

```diff
@@ -0,0 +1,27 @@
+module IrsAttemptsApi
+  class Tracker
+    attr_reader :session_id
+
+    def initialize(session_id:)
+      @session_id = session_id
+    end
+
+    def track_event(event_type, metadata = {})
+      return unless IdentityConfig.store.irs_attempt_api_enabled
+
+      jti, jwe = IrsAttemptsApi::EncryptedEventTokenBuilder.new(
+        event_type: event_type,
+        session_id: session_id,
+        occurred_at: Time.zone.now,
+        event_metadata: metadata,
+      ).build_event_token
+      redis_client.write_event(jti, jwe)
+    end
```

Comment:
does this method need to return the `jti`? Otherwise how will we find the event we wrote?
```suggestion
      redis_client.write_event(jti, jwe)
      jti
    end
```

Reply:
It probably doesn't hurt, but we should need the JTI after this point. The events can be delivered out of order so we can just do a pscan for keys and return events off the top